solvers/ceres_solver.cpp

```diff
@@ -22,24 +22,55 @@ CeresSolver::CeresSolver()
 }
 
 /*****************************************************************************/
-void CeresSolver::Configure(rclcpp::Node::SharedPtr node)
+void CeresSolver::Configure(rclcpp_lifecycle::LifecycleNode::SharedPtr node)
 /*****************************************************************************/
 {
-  node_ = node;
+  logger_ = node->get_logger();
 
   std::string solver_type, preconditioner_type, dogleg_type,
     trust_strategy, loss_fn, mode;
-  solver_type = node->declare_parameter("ceres_linear_solver",
-      std::string("SPARSE_NORMAL_CHOLESKY"));
-  preconditioner_type = node->declare_parameter("ceres_preconditioner",
-      std::string("JACOBI"));
-  dogleg_type = node->declare_parameter("ceres_dogleg_type",
-      std::string("TRADITIONAL_DOGLEG"));
-  trust_strategy = node->declare_parameter("ceres_trust_strategy",
-      std::string("LM"));
-  loss_fn = node->declare_parameter("ceres_loss_function",
-      std::string("None"));
-  mode = node->declare_parameter("mode", std::string("mapping"));
+  if (!node->has_parameter("ceres_linear_solver")) {
```

Comment:
nav2_util's declare if not already declared would make this all much more ergonomic